crypto/evp/bio_b64.c

```diff
@@ -209,13 +208,11 @@ static int b64_read(BIO *b, char *out, int outl)
                 k = EVP_DecodeUpdate(ctx->base64,
                                      (unsigned char *)ctx->buf,
                                      &num, p, q - p);
-                if ((k <= 0) && (num == 0) && (ctx->start))
+                if (k <= 0 && num == 0 && ctx->start) {
                     EVP_DecodeInit(ctx->base64);
-                else {
-                    if (p != (unsigned char *)
-                        &(ctx->tmp[0])) {
-                        i -= (p - (unsigned char *)
-                              &(ctx->tmp[0]));
+                } else {
+                    if (p != (unsigned char *)ctx->tmp) {
+                        i -= (p - (unsigned char *)ctx->tmp);
```

Comment:
This doesn't need the outer set of parenthesis.

Reply:
Right. Though I thought it may be good to keep the parentheses here for clarity with the two  `-` being involved, I removed them.

Reply:
Also got rid of most of those ugly `(unsigned char*)` type casts.